src/main/java/gregtech/common/pipelike/fluidpipe/BlockFluidPipe.java

```diff
@@ -110,36 +110,21 @@ public boolean isHoldingPipe(EntityPlayer player) {
     }
 
     @Override
-    public void onEntityCollision(World worldIn, @Nonnull BlockPos pos, @Nonnull IBlockState state, @Nonnull Entity entityIn) {
+    public void onEntityCollision(@Nonnull World worldIn, @Nonnull BlockPos pos, @Nonnull IBlockState state, @Nonnull Entity entityIn) {
         if (worldIn.isRemote) return;
-        if (entityIn instanceof EntityLivingBase && entityIn.world.getTotalWorldTime() % 20 == 0L) {
+        if (entityIn instanceof EntityLivingBase && worldIn.getTotalWorldTime() % 10 == 0) {
             EntityLivingBase entityLiving = (EntityLivingBase) entityIn;
             TileEntityFluidPipe pipe = (TileEntityFluidPipe) getPipeTileEntity(worldIn, pos);
-            if(!(pipe instanceof TileEntityFluidPipeTickable))
+            if (!(pipe instanceof TileEntityFluidPipeTickable))
                 return;
-            List<Integer> temps = new ArrayList<>();
+
+            // apply temperature damage for each pipe
             for (FluidTank tank : ((TileEntityFluidPipeTickable) pipe).getFluidTanks()) {
                 if (tank.getFluid() != null && tank.getFluid().amount > 0) {
-                    temps.add(tank.getFluid().getFluid().getTemperature(tank.getFluid()));
+                    int temperature = tank.getFluid().getFluid().getTemperature(tank.getFluid());
+                    EntityDamageUtil.applyTemperatureDamage(entityLiving, temperature, 1.0F, 5);
                 }
             }
```

Comment:
Adding the temperatures of all fluids as damage seems brutal. I used the average temperature before.